pkg/controllermanager/controller/shoot/shoot_control_reconcile.go

```diff
@@ -411,15 +436,21 @@ func (c *Controller) updateShootStatusReconcileSuccess(o *operation.Operation, o
 	return err
 }
 
-func (c *Controller) updateShootStatusReconcileError(o *operation.Operation, operationType gardencorev1alpha1.LastOperationType, lastError *gardencorev1alpha1.LastError) error {
+func (c *Controller) updateShootStatusReconcileError(o *operation.Operation, operationType gardencorev1alpha1.LastOperationType, description string, lastErrors ...gardencorev1alpha1.LastError) error {
 	var (
 		state         = gardencorev1alpha1.LastOperationStateFailed
-		description   = lastError.Description
 		lastOperation = o.Shoot.Info.Status.LastOperation
 		progress      = 1
 		willRetry     = !utils.TimeElapsed(o.Shoot.Info.Status.RetryCycleStartTime, c.config.Controllers.Shoot.RetryDuration.Duration)
 	)
 
+	// TODO: Remove this after LastError is removed from the ShootStatus API
+	var codes []gardencorev1alpha1.ErrorCode
+	for _, lastErr := range lastErrors {
+		codes = append(codes, lastErr.Codes...)
+	}
+	lastError := gardencorev1alpha1helper.LastError(description, codes...)
```

Comment:
Okay, so `status.lastError` contains all errors, and `status.lastErrors[]` contains also all errors, however, separated into a list with dedicated entries, right?

Reply:
I've seen this used in [this errors package][1] and some other places. 
The idea is that you force extracting the `id` only from the `GetID(err error)` method.

In our case this might be a bit too much limitation as there isn't any special logic (as in the cycle in [the error package][1])

[1]: https://github.com/pkg/errors/blob/master/errors.go#L270

Reply:
Yes, but there is one difference.  The error descriptions in status.lastErrors[].Descriptions do not contain the following text:
`flow %q encountered task errors` which is present in status.LastError.Description and status.LastOperation.Description